File: apps/ios/PagApp/PagApp/Models/AuthUser.swift
import Foundation
import FirebaseAuth

public struct AuthUser: Identifiable, Equatable {
    public var id: String { uid }
    public let uid: String
    public let email: String?
    public let displayName: String?
    public let photoURL: URL?
    public let provider: String

    public init(uid: String, email: String? = nil, displayName: String? = nil, photoURL: URL? = nil, provider: String = "firebase") {
        self.uid = uid
        self.email = email
        self.displayName = displayName
        self.photoURL = photoURL
        self.provider = provider
    }

    public init(user: User) {
        self.uid = user.uid
        self.email = user.email
        self.displayName = user.displayName
        self.photoURL = user.photoURL
        self.provider = user.providerData.first?.providerID ?? "firebase"
    }
}

File: apps/android/app/src/main/java/com/pagapp/pag/models/AuthUser.kt
package com.pagapp.pag.models

import com.google.firebase.auth.FirebaseUser

data class AuthUser(
    val uid: String,
    val email: String? = null,
    val displayName: String? = null,
    val photoUrl: String? = null,
    val provider: String = "firebase"
) {
    companion object {
        fun fromFirebaseUser(user: FirebaseUser): AuthUser {
            val primaryProvider = user.providerData.firstOrNull()?.providerId ?: "firebase"
            return AuthUser(
                uid = user.uid,
                email = user.email,
                displayName = user.displayName,
                photoUrl = user.photoUrl?.toString(),
                provider = primaryProvider
            )
        }
    }
}
